app/src/main/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonFilters.kt

```diff
@@ -0,0 +1,144 @@
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+internal fun LessonLanguageFilter(modifier: Modifier = Modifier, viewModel: LessonsViewModel = viewModel()) {
+    val context = LocalContext.current
+    val selectedLanguage by viewModel.selectedLanguage.collectAsState(initial = Language(Locale.getDefault()))
+
+    val lessonLanguages by viewModel.filteredLanguages.collectAsState(emptyList())
+    var expanded by remember { mutableStateOf(false) }
+
+    ElevatedButton(
+        onClick = { expanded = true },
+        modifier = modifier.semantics { role = Role.DropdownList }
+    ) {
+        Text(
+            selectedLanguage.getDisplayName(context, LocalAppLanguage.current),
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier.weight(1f),
+        )
+        ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded)
+        LazyDropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false },
+            modifier = Modifier.sizeIn(maxHeight = DROPDOWN_LESSON_MAX_HEIGHT, maxWidth = DROPDOWN_LESSON_MAX_WIDTH)
+        ) {
+            stickyHeader {
+                Surface(color = MaterialTheme.colorScheme.surface) {
+                    val query by viewModel.query.collectAsState("")
+                    SearchBar(
+                        query = query,
+                        onQueryChange = { change ->
+                            viewModel.query.value = change
+                        },
+                        onSearch = {},
+                        active = false,
+                        onActiveChange = {},
+                        colors = GodToolsTheme.searchBarColors,
+                        placeholder = {
+                            Text(stringResource(R.string.language_settings_downloadable_languages_search))
+                        },
+                        content = {},
+                        modifier = Modifier
+                            .padding(horizontal = 8.dp)
+                            .fillMaxWidth()
+                            .wrapContentWidth()
+                    )
+                }
+            }
+
+            items(lessonLanguages, key = { (lang) -> lang.code }) { (lang, count) ->
+                FilterMenuItem(
+                    label = { LanguageName(lang) },
+                    supportingText = pluralStringResource(
+                        R.plurals.dashboard_lessons_section_filter_available_lessons,
+                        count,
+                        count,
+                    ),
+                    onClick = {
+                        expanded = false
+                        viewModel.updateSelectedLanguage(lang)
+                    },
+                    modifier = Modifier.animateItemPlacement()
+                )
+                HorizontalDivider(Modifier.padding(start = ListItemStartPadding, end = ListItemStartPadding))
```

Comment:
you now have an extra divider at the end of the popup.
![Screenshot 2024-08-21 at 1 50 48 PM](https://github.com/user-attachments/assets/2828916c-8b85-4a92-a215-1594d78743bf)

Look into using `itemsIndexed` to solve the problem